### ISSUE TITLE
How to set YetiRank mode to NDCG ?

### ISSUE THREAD

**Issue by glmn** [catboost/catboost#2712]
Problem: Don't know how to set mode for YetiRank loss function.

Screenshot from [documentation](https://catboost.ai/en/docs/concepts/loss-functions-ranking#YetiRank): 
![image](https://github.com/user-attachments/assets/9ec46327-0ed5-4c65-b731-e4a24de70ccc)

```python
default_parameters = {
    'iterations': 1000,
    'verbose': False,
    'learning_rate': 0.01,
    'depth':12,
    'loss_function': 'YetiRank:mode=ERR',
    'task_type': 'GPU'
}
model = CatBoostRanker(**default_parameters).fit(train, eval_set=test, plot=True)
```

If I try to set `loss_function` to `YetiRank:mode=NDCG` then it doesn't have any effect. It still works like default `Classic`.
Event If I set it to `YetiRank:mode=TESTERROR` it does not throw an error about unknown mode and still works as `Classic`.

Am I miss something or missunderstand documentation?

CatBoost version: 1.2.5
OS: Windows 10

**Comment by andrey-khropov** [comment]
> Am I miss something or missunderstand documentation?

As it is said in the documentation that you cite:

> Non-Classic modes are supported only on CPU

But you use GPU here so the only supported `Classic` mode is used.

But CatBoost should validate metric parameters more thoroughly in this case and print a meaningful error. We'll fix that.

**Comment by glmn** [comment]
> Non-Classic modes are supported only on CPU

I've tested it also with `'task_type': 'CPU'`.
No effect at all.
